compiler/version_check.go

```diff
@@ -1,9 +1,9 @@
-// +build go1.11
-// +build !go1.12
+// +build go1.12
+// +build !go1.13
 
 package compiler
 
-const ___GOPHERJS_REQUIRES_GO_VERSION_1_11___ = true
+const ___GOPHERJS_REQUIRES_GO_VERSION_1_12___ = true
 
 // Version is the GopherJS compiler version string.
-const Version = "1.11-2"
+const Version = "1.12-wip"
```

Comment:
This shouldn't be `wip`.

Reply:
Done.

Reply:
@neelance Why not? I think it should be `wip` while this PR is in active development and review, and changed to `1.12-1` only when it's about to be merged.

That way, when you run `gopherjs version` and get `1.12-1`, it's guaranteed to be the final version and not an older in-development one.

It's also how we've done the last 3+ releases. See https://github.com/gopherjs/gopherjs/pull/853/commits/8b6e87829e4f5245fde933d574a6cd89ec008d9b for example.

Reply:
Sooner or later we'll change `1.12` before merging, so I think changing this now should be fine.

Reply:
I would prefer to see the `-wip` suffix when the version is in development. Removing it and replacing it with `-1` should be done when the merge is happening.

This is consistent with how the version of Go toolchain gets bumped only when the release is happening, not sooner. E.g., see https://github.com/golang/go/commit/05e77d41914d247a1e7caf37d7125ccaa5a53505.

Reply:
Just to confirm, will we squash these commits when merging?

Reply:
No, we've never done that in the past. Squashing would create a giant commit without a meaningful commit message.

It has always been a merge commit, typically after rebasing the PR on top of latest `master` branch and cleaning up its history a bit.

See previous PRs and their merge commits:

- Go 1.11 support - PR #853 - merge commit https://github.com/gopherjs/gopherjs/commit/0210a2f0f73c96103378b0b935f39868e5731809
- Go 1.10 support - PR #755 - merge commit - https://github.com/gopherjs/gopherjs/commit/4e9f423448a0b0462d41b077c43bb9f2a5d103c6
- Go 1.9 support - PR #651 - merge commit https://github.com/gopherjs/gopherjs/commit/f7c56533adfac99e12ccdccfec04a466d2a778da
- Go 1.8 support - PR #552 - merge commit https://github.com/gopherjs/gopherjs/commit/b9bcb1da229a59cc1e1d168401662cb6450aae08
- Go 1.7 support - PR #495 - merge commit https://github.com/gopherjs/gopherjs/commit/b33dd36ad3f3b02fc01a860f94becdb4e9a95ddc
- Go 1.6 support - PR #404 - merge commit https://github.com/gopherjs/gopherjs/commit/f8510c9aef19750ec5a1178b33932a7f9defd197

GopherJS versions were proposed in issue #500 and implemented as of GopherJS 1.7-1 (after it was already on `master`). So if you look at PRs for 1.8 and newer, you'll find a "GopherJS 1.x-wip" -> "GopherJS 1.x-1" bump commit at the very end. E.g.:

- https://github.com/gopherjs/gopherjs/pull/853/commits/8b6e87829e4f5245fde933d574a6cd89ec008d9b - bump version from 1.11-wip to 1.11-1
- https://github.com/gopherjs/gopherjs/pull/755/commits/4ebbd33f7fb1fca27e0a0a413c2729ae8cb4d958 - bump version from 1.10-wip to 1.10-1
- https://github.com/gopherjs/gopherjs/pull/651/commits/fe63fb1b9957f490d5011c8bb0a254db64e6a138 - bump version from 1.9-wip to 1.9-1
- https://github.com/gopherjs/gopherjs/pull/552/commits/0ff5023f798422e0389990d6edff2b06e7861e4d - notable exception, but it's because the history of the `go1.8` branch was re-written before merged and the wip->1 bump was squashed into the initial version commit

Reply:
Done.